tests/zfs-tests/tests/functional/fault/auto_spare_shared.ksh

```diff
@@ -0,0 +1,116 @@
+#!/bin/ksh -p
+#
+# CDDL HEADER START
+#
+# This file and its contents are supplied under the terms of the
+# Common Development and Distribution License ("CDDL"), version 1.0.
+# You may only use this file in accordance with the terms of version
+# 1.0 of the CDDL.
+#
+# A full copy of the text of the CDDL should have accompanied this
+# source.  A copy of the CDDL is also available via the Internet at
+# http://www.illumos.org/license/CDDL.
+#
+# CDDL HEADER END
+#
+
+#
+# Copyright 2018, loli10K <ezomori.nozomu@gmail.com>. All rights reserved.
+#
+
+. $STF_SUITE/include/libtest.shlib
+. $STF_SUITE/include/math.shlib
+. $STF_SUITE/tests/functional/fault/fault.cfg
+
+#
+# DESCRIPTION:
+# Spare devices (both files and disks) can be shared among different ZFS pools.
+#
+# STRATEGY:
+# 1. Create two pools
+# 2. Add the same spare device to different pools
+# 3. Inject IO errors with a zinject error handler
+# 4. Start a scrub
+# 5. Verify the ZED kicks in a hot spare and check pool/device status
+# 6. Clear the fault
+# 7. Verify the hot spare is available and check pool/device status
+#
+
+verify_runnable "both"
+
+if is_linux; then
+	load_scsi_debug $SDSIZE 1 1 1 '512b'
```

Comment:
Should it be `$SPA_MINDEVSIZE` instead of `$SDSIZE` here?

Reply:
`$SPA_MINDEVSIZE` is a number in bytes, scsi_debug wants the size parameter in MB (default `$SDSIZE` is 256). I am going to copy "auto_spare_ashift.ksh":

https://github.com/zfsonlinux/zfs/blob/b8a90418f3a9c23b89c5d2c729a4dd0fea644508/tests/zfs-tests/tests/functional/fault/auto_spare_ashift.ksh#L74

Reply:
Ok, add the comment from there in too:
```
# 2. Add one 512e spare device (4Kn would generate IO errors on replace)
# NOTE: must be larger than the existing 512b devices, add 32m of fudge
```